src/openforms/config/models.py

```diff
@@ -661,19 +661,35 @@ def as_dict(self):
 class CSPSettingManager(models.Manager.from_queryset(CSPSettingQuerySet)):
     @transaction.atomic
     def set_for(
-        self, obj: models.Model, settings: list[tuple[CSPDirective, str]]
+        self,
+        obj: models.Model,
+        settings: list[tuple[CSPDirective, str]],
+        identifier: str = "",
     ) -> None:
         """
-        Deletes all the connected csp settings and creates new ones based on the new provided data.
+        Deletes all the connected CSP settings and creates new ones based on the new provided data.
+
+        :param obj: The configuration model providing this CSP entry.
+        :param settings: A two-tuple containing values used to create the underlying ``CSPSetting`` model.
+        :param identifier: An optional string to further identify the source of this CSP entry.
         """
         instances = [
-            CSPSetting(content_object=obj, directive=directive, value=value)
+            CSPSetting(
+                content_object=obj,
+                directive=directive,
+                value=value,
+                identifier=identifier or "",
+            )
             for directive, value in settings
         ]
 
-        CSPSetting.objects.filter(
-            content_type=get_content_type_for_model(obj), object_id=str(obj.id)
-        ).delete()
+        predicate = {
+            "content_type": get_content_type_for_model(obj),
+            "object_id": str(obj.id),
+        }
+        if identifier is not None:
+            predicate["identifier"] = identifier
```

Comment:
this condition can/should be changed now into

```suggestion
        if identifier:
            predicate["identifier"] = identifier
```
or just drop it alltogether, IMO I think you can just pass in all three filter params directly to `.filter(...)`, the result is the same